### PR TITLE
Fix #6608 normalized job_driver's agent life cycle code

### LINKED ISSUE #6572: Improve logging and failure cases related to sirepo.status
This shouldn't throw an exception like this. It should dump the message:

```txt
20:28:36 sirepo/status.py", line 88, in _run_tests
20:28:36 d = r.nextRequest
20:28:36 pykern/pkcollections.py", line 68, in __getattr__
20:28:36 raise AttributeError(
20:28:36 AttributeError: 'PKDict' object has no attribute 'nextRequest'
```

This switches the args. It should also dump the entire message, not just the opName:

```txt
17:17:11 /sirepo/job_driver/__init__.py:330:_receive DockerDriver(a=TiMy k=sequential u=<snip> []) not pending opName=<long opid> o=run
```

### LINKED ISSUE #6608: job_driver.kill is async, which can clear an unmanaged op
At the time of `send` the op being sent should be in the queue, and here you see the queue is empty:

`17:17:03 /sirepo/job_driver/__init__.py:194:send DockerDriver(a=TiMy k=sequential u=uid1 [<EMPTY>]) _Op(run, GgOX)`

A normal send looks like this:

`17:13:41 /sirepo/job_driver/__init__.py:194:send DockerDriver(a=FnzR k=parallel u=uid1 [_Op(run, ZgIU)]) _Op(run, ZgIU)`

At init, it is added:

`17:17:00 /sirepo/job_supervisor.py:1094:__init__ _Op(run, GgOX)`

A `kill_if_idle` starts for `TiMy` agent before `GgOX` is received,
which adds `GgOX` to the driver's (`TiMy`) ops:

```txt
17:16:59 /sirepo/job_driver/__init__.py:382:_kill_if_idle DockerDriver(a=TiMy k=sequential u=uid1 [])
17:16:59 /sirepo/job_driver/docker.py:121:kill DockerDriver(a=TiMy k=sequential u=uid1 []) cid=e27a09ba43de
17:16:59 /sirepo/job_supervisor.py:238:receive ServerReq(api_runSimulation, uid1-sid1-initialIntensityReport)
17:17:00 /sirepo/job_supervisor.py:1094:__init__ _Op(run, GgOX)
```

The `websocket_on_close` happens which removes `GgOX` from the list of
ops, because the agent is being killed:

```txt
17:17:00 /sirepo/job_driver/__init__.py:210:websocket_on_close DockerDriver(a=TiMy k=sequential u=uid1 [_Op(run, GgOX)])
17:17:00 /sirepo/job_driver/__init__.py:117:free_resources DockerDriver(a=TiMy k=sequential u=uid1 [_Op(run, GgOX)]) internal_error=None
17:17:00 /sirepo/job_driver/__init__.py:282:_agent_start DockerDriver(a=TiMy k=sequential u=uid1 []) _Op(run, GgOX) await _do_agent_start
```

The `free_resources` removed `GgOX` from the queue (note the empty
`[]`). `_agent_start` has the `op` on the stack so that's why it is
printed, but the queue is empty (op is dangling).

At this point there are no ops (empty `[]`). While the driver is
waiting for the kill to complete, an op comes in (`GgOX` is
created). However, the
[job_driver.kill](https://github.com/radiasoft/sirepo/blob/dc75c77b1a5ab9c0abcf1a664af0e3317cc14949/sirepo/job_driver/__init__.py#L383)
has an await. The agent is still "active" so the op get queued, added
to the live driver.

Need to figure out locking for this scenario.

### LINKED ISSUE #6613: job_driver._agent_start_delay is global but should be local
This is only for internal_test, and when a specific value is passed for a specific simulation. It should be bound to something other than the driver, because the driver is a persistent object.

### PR DESCRIPTION
**Pull request by robnagler**
- Changed some conditionals to assertions to simplify code
- Locking around start and end of the agent life (_agent_life_change_lock)
- Ops are bound to driver in prepare_send, was too early in agent_start
- Fix #6613 _agent_start_delay is local to op
- Fix #6572 added more logging, fixed some messages, and normalized others
- Cleaned up some naming (opId > op_id when not in msg)
- Cascade exceptions into op.internal_error for better logging
- Improved error handling/logging in sirepo.status

### CODE REVIEW

**Review by e-carlin — APPROVED** [review]
Looks good! Two nits

File: sirepo/job_driver/__init__.py [hunk 1]
```diff
@@ -323,64 +290,112 @@ def _receive(self, msg):
...
+    def _agent_receive_error(self, msg):
         # TODO(robnagler) what does this mean? Just a way of logging? Document this.
         pkdlog("{} msg={}", self, msg)
 
+    async def _agent_start(self, op):
+        if self._websocket_ready_timeout:
+            return
+        try:
+            async with self._agent_life_change_lock:
+                # POSIT: we do not have to raise Awaited(), because
```
[the remark this answers is not in the document]
Comment (robnagler) [reply sirepo/job_driver/__init__.py:330]: I'm going to improve the documentation for Awaited. I think it should be "WaitedForResources". I think the problem being solved is that the agent may terminate during the acquisition of resources, and agent_ready has to be rechecked. The Slots (via SlotProxy) are not released so [they don't await](https://github.com/radiasoft/sirepo/blob/50590584e4398f7fe1d691a7d0aebbc03a8d44a7/sirepo/job_supervisor.py#L107) and which makes those checks non-blocking (always). This must be true for locks, too, which yield the processor if there was a wait. 

Perhaps what needs to happen is that agent_ready is checked twice: before slots_ready and after.

This comment is about starting the agent which has already started, and that's why it is not correct.

File: sirepo/job_driver/__init__.py [hunk 2]
```diff
@@ -323,64 +290,112 @@ def _receive(self, msg):
...
         """
-        self._agent_starting_done()
+        self._websocket_ready_timeout_cancel()
         if self._websocket:
-            if self._websocket != msg.handler:
-                pkdlog("{} new websocket", self)
-                # New _websocket so bind
-                self.free_resources()
-        self._websocket = msg.handler
+            assert self._websocket == msg.handler, f"incoming msg.content={msg.content}"
```
[the remark this answers is not in the document]
Comment (robnagler) [reply sirepo/job_driver/__init__.py]: 739c693

File: sirepo/job_driver/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (robnagler) [reply sirepo/job_driver/__init__.py]: This turned into a long discussion. I'll try to summarize in the merge. [667340a](https://github.com/radiasoft/sirepo/pull/6617/commits/667340a4afb69219798545b035ace3650708dea8)

**Pull request merged**